src/v/config/configuration.cc

```diff
@@ -1582,6 +1582,14 @@ configuration::configuration()
       "violations. Normally, this options should be disabled.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       false)
+  , cloud_storage_disable_metadata_consistency_checks(
+      *this,
+      "cloud_storage_disable_metadata_consistency_checks",
+      "Disable all metadata consistency checks. This will allow redpanda to "
+      "replay logs with inconsistent tiered-storage metadata. Normally, this "
+      "option should be disabled.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      true)
```

Comment:
>Normally, this "
>       "option should be disabled."

and it is enabled?

Reply:
Right, we are disabling it here to allow log replays to proceed even if there's anomalies in it from past versions of redpanda. Evgeny, has another PR that bumps the serde version for the add segment command. We can re-enable it when/if we backport that.

Reply:
Yes, because we want to disable the feature for old clusters. We will eventually flip this when another PR will be merged.